src/tools/perf/lib/ucp_tests.cc

```diff
@@ -366,31 +366,68 @@ class ucp_perf_test_runner {
         return UCS_PTR_BYTE_OFFSET(buffer, length - sizeof(psn_t));
     }
 
+    UCS_F_ALWAYS_INLINE ucs_status_t ucp_perftest_wait_for_completion(
+            ucs_status_ptr_t request, ucp_worker_h worker)
+    {
+        ucs_status_t status;
```

Comment:
first, check if the request is NULL or error. See other places that wait for a request.